src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/PreferMethodReferenceCheck.java

```diff
@@ -0,0 +1,337 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2019 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.checks.coding;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import com.puppycrawl.tools.checkstyle.StatelessCheck;
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.TokenUtil;
+
+/**
+ * <p>
+ * Checks that lambda usage can be replaced with method reference.
+ * </p>
+ * <p>
+ * Rationale: method references are easier to read.
+ * </p>
+ * <p>
+ * To configure the check:
+ * </p>
+ * <pre>
+ * &lt;module name="PreferMethodReference"/&gt;
+ * </pre>
+ * <p>Example of violations:</p>
+ * <pre>
+ * Runnable r = () -&gt; System.out.println(); // violation
+ * BiConsumer&lt;Long, Long&gt; c = (a, b) -&gt; Long.sum(a,b); // violation
+ * </pre>
+ * <p>
+ * Note: This check is not type-aware, so some cases cannot be detected, e.g.
+ * </p>
+ * <pre>
+ * Function&lt;String, Object&gt; f1 =
+ *     a -&gt; (Object) "string".concat(a); // no violation because of cast
+ * </pre>
+ *
+ * @since 8.23
+ */
+@StatelessCheck
+public class PreferMethodReferenceCheck extends AbstractCheck {
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_METHOD_REF = "method.reference";
+
+    /**
+     * List of tokens that is used to check expression.
+     * @see #shouldCheck(DetailAST)
```

Comment:
Please add a blank line before the tags section as suggested in #4356
We did not enforce this right now, but in this PR part of comments follow this rule, and others do not. Please make them all consistent.

Reply:
done